hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/GlobalSearchService.kt

```diff
@@ -0,0 +1,205 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services
+
+import com.google.gson.Gson
+import com.microsoft.applicationinsights.TelemetryClient
+import org.apache.lucene.search.join.ScoreMode
+import org.opensearch.action.search.SearchRequest
+import org.opensearch.action.search.SearchResponse
+import org.opensearch.index.query.BoolQueryBuilder
+import org.opensearch.index.query.QueryBuilders
+import org.opensearch.search.builder.SearchSourceBuilder
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.Pageable
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.search.config.AuthenticationHolder
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.exceptions.BadRequestException
+
+@Service
+class GlobalSearchService(
+  private val searchClient: SearchClient,
+  private val gson: Gson,
+  private val telemetryClient: TelemetryClient,
+  private val authenticationHolder: AuthenticationHolder,
+) {
+  companion object {
+    private val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun findByGlobalSearchCriteria(globalSearchCriteria: GlobalSearchCriteria, pageable: Pageable): Page<Prisoner> {
+    validateSearchForm(globalSearchCriteria)
+    if (globalSearchCriteria.prisonerIdentifier != null) {
+      queryBy(globalSearchCriteria, pageable) { idMatch(it) } onMatch {
+        customEventForFindBySearchCriteria(globalSearchCriteria, it.matches.size)
+        return PageImpl(it.matches, pageable, it.totalHits)
+      }
+    }
+    if (!(globalSearchCriteria.firstName.isNullOrBlank() && globalSearchCriteria.lastName.isNullOrBlank())) {
+      if (globalSearchCriteria.includeAliases) {
+        queryBy(globalSearchCriteria, pageable) { nameMatchWithAliases(it) } onMatch {
+          customEventForFindBySearchCriteria(globalSearchCriteria, it.matches.size)
+          return PageImpl(it.matches, pageable, it.totalHits)
+        }
+      } else {
+        queryBy(globalSearchCriteria, pageable) { nameMatch(it) } onMatch {
+          customEventForFindBySearchCriteria(globalSearchCriteria, it.matches.size)
+          return PageImpl(it.matches, pageable, it.totalHits)
+        }
+      }
+    }
+    return PageImpl(listOf(), pageable, 0L)
+  }
```

Comment:
As this is the top level method for this class it might be worth spending some time making it more readable. For example if you did a `with(globalSearchCriteria) {` you'd lose a lot of noise instantly.

Reply:
good idea - it was just a lift and shift from prisoner-offender-search but the next few bits of work I will be looking at tidying things up